app/models/principal.rb

```diff
@@ -71,8 +71,32 @@ def full_name
     "#{first_name} #{last_name}"
   end
 
+  def next_onboarding_action
+    registered_firms = main_firm_with_trading_names.registered
+
+    if registered_firms.empty?
+      :complete_a_firm
+    elsif needs_advisers?(registered_firms)
+      :complete_an_adviser
+    else
+      :onboarded
+    end
+  end
+
   private
 
+  def needs_advisers?(firm_list)
+     no_remote_firms?(firm_list) && no_advisers_for?(firm_list)
```

Comment:
On a more serious note this logic is, by it's nature, confusing. That is why we've had several long conversations on it, why I have a spreadsheet listing all of the state permutations on my desktop, and why we have close to 400 lines of tests around this. We landed at this implementation being the most descriptive. Happy to change it if a better solution is found though.

We did stick for a while at the negative naming, but the situation we are coding for is this:

* If the principal has _at least once_ **remote** firm then they are `:onboarded` as remote firms are able to be added to the register as long as the firm is complete.
* If they have _no_ **remote** firms then we check to see if they need advisers

We did originally have the method to check for "not remote" being "for in-person", but it was actually misleading as the mind automatically goes to the version that Dan has above, which is actually incorrect. It needs to be _all_ in_person, not _any_ in_person. :)